packages/main/src/index.ts

```diff
@@ -98,33 +73,54 @@ if (import.meta.env.PROD) {
 
 let tray: Tray | null = null;
 
-app.whenReady().then(async () => {
-  // Setup the default tray icon + menu items
-  const animatedTray = new AnimatedTray();
-  tray = new Tray(animatedTray.getDefaultImage());
-  animatedTray.setTray(tray);
-  const trayMenu = new TrayMenu(tray, animatedTray);
-
-  // Start extensions
-  const pluginSystem = new PluginSystem(trayMenu);
-  const extensionLoader = await pluginSystem.initExtensions();
-
-  // Get the configuration registry (saves all our settings)
-  const configurationRegistry = extensionLoader.getConfigurationRegistry();
-
-  // If we've manually set the tray icon color, update the tray icon. This can only be done
-  // after configurationRegistry is loaded. Windows or Linux support only for icon color change.
-  if (!isMac) {
-    const color = configurationRegistry.getConfiguration('preferences').get('TrayIconColor');
-    if (typeof color === 'string') {
-      animatedTray.setColor(color);
+app.whenReady().then(
+  async () => {
+    // We must create the window first before initialization so that we can load the
+    // configuration as well as plugins
+    // The window is hiddenly created and shown when ready
+
+    // Platforms: Linux, macOS, Windows
+    // Create the main window
+    createNewWindow();
+
+    // Platforms: macOS
+    // Required for macOS to start the app correctly (this is will be shown in the dock)
+    // https://www.electronjs.org/docs/latest/api/app#event-activate-macos
+    app.on('activate', createNewWindow);
```

Comment:
IMHO it looks very odd to have the app.on inside a app.whenReady code

Reply:
This helps solve the issue of it not starting gracefully in macOS rather than calling restore / create window multiple times outside of whenReady (how we had it before this PR). This is referenced multiple times within the tutorial / electron website on using activate within whenReady. The reason is that when using it outside of whenReady, it's not cleanly starting up, causing issues with macOS restoring the window.

Here's a list of examples where activate is used within whenReady code:

* Tutorial example: https://www.electronjs.org/docs/latest/tutorial/quick-start#open-a-window-if-none-are-open-macos (which solves the issue I'm fixing regarding minimizing / showing)
* https://www.electronjs.org/docs/latest/#running-examples-with-electron-fiddle (see fiddle example)
* https://www.electronjs.org/docs/latest/tutorial/ipc#pattern-3-main-to-renderer

All of which are implemented because of the macOS issue.

I've gone ahead and added the https://www.electronjs.org/docs/latest/tutorial/quick-start#open-a-window-if-none-are-open-macos link to the comment to help clarify why it may look odd, but follows the same electronjs standards.